be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -811,7 +813,16 @@ Status SegmentWriter::append_block(const vectorized::Block* block, size_t row_po
         }
         RETURN_IF_ERROR(_column_writers[id]->append(converted_result.second->get_nullmap(),
                                                     converted_result.second->get_data(), num_rows));
+        for (size_t row_id = 0; row_id < block->rows(); ++row_id) {
+            const auto& stringRef = block->get_by_position(cid).column->get_data_at(row_id);
```

Comment:
`get_data_at()` this function is used in ColumnString, ColumnFixedString, ColumnVector, not support in ColumnArray|ColumnMap...,and should be pair with insert_data.